### PR TITLE
Create endpoints to check service status

### PR DESCRIPTION
**Pull request by nabanitadewan**
Created endpoints which query count of rows in product and article tables to check if the service is running or not.

This is my first time contributing to OSS , would love to hear your feedback . :)

### CODE REVIEW

**Comment by nabanitadewan** [comment]
@tiagostutz please review my PR.

File: api-backend/src/routes/article.ts [hunk 1]
```diff
@@ -16,7 +17,15 @@ export default {
      * - 204 : the service is OK
      * - 404 : the service is Degraded
      */
-    app.get(`/${prefix}/health`, async (_, res) => res.status(204).send());
+     app.get(`/${prefix}/health`, async (_, res) => {
+      try {
+        await checkArticleHealth();
+        res.status(201).send("ok");
```
Comment (tiagostutz) [inline api-backend/src/routes/article.ts]: Change status to 200. The "OK" status is 200.
Check this https://httpstatuses.com

File: api-backend/src/routes/article.ts [hunk 2]
```diff
@@ -16,7 +17,15 @@ export default {
...
+      } catch (error: any) {
+        log.info(error);
+        res.status(404).send('Degraded');
```
Comment (tiagostutz) [inline api-backend/src/routes/article.ts]: If there was an error, the status should be 5xx. As this is an unexpected error, use 500 as status.

File: api-backend/src/routes/product.ts [hunk 3]
```diff
@@ -11,7 +11,15 @@ export default {
      * - 204 : the service is OK
      * - 404 : the service is Degraded
      */
-    app.get(`/${prefix}/health`, async (_, res) => res.status(204).send());
+     app.get(`/${prefix}/health`, async (_, res) => {
+      try {
+        await  checkProductHealth();
+        res.status(201).send("ok");
```
Comment (tiagostutz) [inline api-backend/src/routes/product.ts]: 200 as status

File: api-backend/src/routes/product.ts [hunk 4]
```diff
@@ -11,7 +11,15 @@ export default {
...
+      } catch (error: any) {
+        log.info(error);
+        res.status(404).send('Degraded');
```
Comment (tiagostutz) [inline api-backend/src/routes/product.ts]: 500 as status

File: api-backend/src/services/article/tests/article.test.ts [hunk 5]
```diff
@@ -84,4 +84,12 @@ describe('Testing CRUD Operations', () => {
     expect(resultGet.error).toBeNull();
     expect(resultGet.article?.identification).toBe(identificationMockBigInt);
   });
+
+  /**
+   * queries the number of rows to check db connection
+   */
+  test('Attempt to query count to check db connection', async () => {
+    const result = await checkArticleHealth();
+    expect(result).not.toBeNull();
```
Comment (tiagostutz) [inline api-backend/src/services/article/tests/article.test.ts:94]: Add one more assert, verify that the status code should be equal to 200

File: api-backend/src/services/product/tests/product.test.ts [hunk 6]
```diff
@@ -312,4 +312,12 @@ describe('Testing Product relationship with Article', () => {
     expect(resultGet.products?.length).toBe(0);
     expect(resultGet.error).toBeNull();
   });
+
+  /**
+   * queries the number of rows to check db connection
+   */
+  test('Attempt to query count to check db connection', async () => {
+    const result = await checkProductHealth();
+    expect(result).not.toBeNull();
```
Comment (tiagostutz) [inline api-backend/src/services/product/tests/product.test.ts:322]: Add assert on the status

**Comment by tiagostutz** [comment]
Hi @nabanitadewan !! Nice to receive your PR, good job!
I did the review and asked for some changes. If you have any doubts, please let me know. 🤓

**Comment by nabanitadewan** [comment]
Hi @tiagostutz , I don't see changes requested.

**Comment by tiagostutz** [comment]
Click on the `Files changed` tab and scroll down over the codes and check that there are inline changing requests, like this:

Screen Shot 2021-11-15 at 22 22 49

**Comment by nabanitadewan** [comment]
@tiagostutz I am not sure if this is because of rights issue, but I still can't see change requests. So taking the screenshot as reference, I have changed Http status code to 200. Could you also paste other change request's screenshot?
image

**Review by tiagostutz — CHANGES_REQUESTED** [review]

**Comment by tiagostutz** [comment]
@nabanitadewan my bad, I had forgot to push "Submit review". Now it should be visible to you 😄

**Comment by nabanitadewan** [comment]
@tiagostutz . I have made the changes as requested. Could you please review it?

**Review by tiagostutz — APPROVED** [review]

**Comment by tiagostutz** [comment]
Great job @nabanitadewan ! Thanks a lot for your contribution on this. 🥇

**Pull request merged**